File: src/main/java/com/core/base/corebase/domain/review/Review.java
package com.core.base.corebase.domain.review;

import com.core.base.corebase.domain.review.code.ReviewState;
import com.core.base.corebase.domain.user.Member;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "review")
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String state;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "review_member_id", nullable = false)
    private ReviewMember reviewMember;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id", nullable = false)
    private Member member;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "review_id", nullable = false)
    private Review review;

    @OneToMany(mappedBy = "review", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ReviewAnswer> answers = new ArrayList<>();

    protected Review() {
        // For JPA
    }

    public Review(ReviewMember reviewMember, Member member, Review review, ReviewState state) {
        this.reviewMember = reviewMember;
        this.member = member;
        this.review = review;
        this.state = state != null ? state.name() : ReviewState.BEFORE.name();
    }

    // Helper methods for bidirectional relationship
    public void addAnswer(ReviewAnswer answer) {
        answers.add(answer);
        answer.setReview(this);
    }

    public void removeAnswer(ReviewAnswer answer) {
        answers.remove(answer);
        answer.setReview(null);
    }

    public ReviewState getState() {
        if(this.state == null) {
            return null;
        }
        return ReviewState.valueOf(state);
    }
}

File: src/main/kotlin/com/core/base/corebase/domain/review/Review.kt
package com.core.base.corebase.domain.review

import com.core.base.corebase.domain.review.code.ReviewState
import com.core.base.corebase.domain.user.Member
import jakarta.persistence.*

@Entity(name = "review")
class Review (
   reviewMember : ReviewMember, //ReviewMember
   member: Member, // 평가자 member ID
   review : Review, //Review 넣을지 말지 고민됨.
   state: ReviewState = ReviewState.BEFORE
){
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0L

    @Enumerated(EnumType.STRING)
    var state = state; protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "review_member_id", nullable = false)
    var reviewMember: ReviewMember = reviewMember; protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id", nullable = false)
    var member: Member = member; protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "review_id", nullable = false)
    var review: Review = review; protected set

    @OneToMany(mappedBy = "review", cascade = [CascadeType.ALL])
    protected val mutableReviewAnswers: MutableList<ReviewAnswer> = mutableListOf()
    val answers: List<ReviewAnswer> get() = mutableReviewAnswers.toList()

}
